crypto/bio/bio_print.c

```diff
@@ -706,9 +706,7 @@ fmtfp(char **sbuffer,
         fconvert[fplace++] = "0123456789"[fracpart % 10];
         fracpart = (fracpart / 10);
     }
-
-    if (fplace == sizeof(fconvert))
-        fplace--;
+  
```

Comment:
This adds an end-of-line whitespace characters.

Reply:
Yes, this code should indeed add a space at the end of the string, however in this case it would be unachievable. 
I think this code illustrates it:
`#include <openssl/evp.h>
#include <openssl/modes.h>
#include <openssl/aes.h>
// gcc -g test.c -Ibuild/include -Lbuild -Lbuild/lib64 -lssl -lcrypto
// example of static analyzer triggering
void my_block128(const unsigned char in[16], unsigned char out[16], const void *key)
{
        const unsigned char *k = (const unsigned char*) key;
        for(int i = 0; i < 16; ++i)
        {
                out[i] = in[i] ^ k[i];
        }
}
void encrypt_data(const unsigned char *plaintext, unsigned char *ciphertext,
                  size_t plaintext_len, const unsigned char *key,
                  const unsigned char *iv) {
        CRYPTO_cbc128_encrypt(plaintext, ciphertext, plaintext_len, key, iv, my_block128);
}
int main() {
    unsigned char plaintext[] = "AAAAAAAAAAAAAAAAA";
    unsigned char key[] = "0123456789abcdef";
    unsigned char iv[] = "fedcba9876543210";
    unsigned char ciphertext[128];

    /* Encrypt the data */
    encrypt_data(plaintext, ciphertext, strlen(plaintext), key, iv);

    /* Print the encrypted data */
    printf("Ciphertext: ");
    for(int i = 0; i < strlen(plaintext); i++) {
        printf("%02x", ciphertext[i]);
    }
    printf("\n");

    return 0;
}`

Reply:
This is a misunderstanding - I am talking about the empty line added in the source code. Please drop that empty line.